### ISSUE TITLE
Admin area crashes with identical property names between pages

### ISSUE THREAD

**Issue by ghost** [sulu/sulu#6610]
| Q | A
| --- | ---
| Sulu Version | 2.4.2
| PHP Version | 8.1.4
| DB Version | MySQL 8.0
| Browser Version | Firefox 100, Chrome 100

# Actual Behavior

When you create two page templates that contain an identically named block, the admin area crashes after template change.

https://user-images.githubusercontent.com/47974249/168613968-312cd02f-37f1-46c7-ae97-128c3fc09cfd.mov

# Expected Behavior

Admin area should not crash.

# Steps to Reproduce

Template 1:

```xml
        <block name="content" default-type="foo" minOccurs="1">
            <types>
                <type name="foo">
                    <properties>
                        <property name="title" type="text_line">

                        </property>
                    </properties>
                </type>
            </types>
        </block>
```

Template 2:

```xml
        <block name="content" default-type="bar" minOccurs="1">
            <types>
                <type name="bar">
                    <properties>
                        <property name="title" type="text_line">

                        </property>
                    </properties>
                </type>
            </types>
        </block>
```

I have built a small example based on the Sulu demo, which you can also see in the video.

https://github.com/flogado/sulu-demo/commit/5db18d32ae9ebd830511ba02c3e00ea3f9fb6373

# Possible Solutions

I think the problem has something to do with trying to put the old state on the new site. This does not work if the new block does not contain the same types.

**Comment by alexander-schranz** [comment]
@flogado Thx for reporting. The error looks for me like the `defaultType` fallback is not correctly handled. The logic to handle this should be here: https://github.com/sulu/sulu/blob/c298925a6815803b5f94ca80be3ffce1c939615f/src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/FieldBlocks.js#L47 do you want to have a look at it?

**Comment by ghost** [comment]
I have tried to get to the bottom of the error. Unfortunately without success. However, I can say that the error is already thrown before the function `componentDidUpdate`.

We will add a workaround in the form of a template without properties for our project. But this is not really a long term solution either.

**Comment by alexander-schranz** [comment]
Okay that is really strange. Thank you for reporting, definitly the issue need to be fixed in the core. But good if you have a workaround which currently works for you.

**Comment by niklasnatter** [comment]
Hey, thanks for reporting! I think the crash is not caused by the identical property names, but by the `resource_locator` property. I think the property causes problems when switching the template of a page that is not yet created. I am not sure yet if this happens for al templates or only if the template contains some special properties.

**Comment by niklasnatter** [comment]
Fixed in #6667
